source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -302,6 +302,16 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       config_->httpContext().codeStats().chargeResponseStat(info);
     }
 
+    // Pass requirement for response code override as a header
```

Comment:
The ext_authz configuration and state aren't all available in the sendLocalReply() where this is being used.   I agree that the header indirection is a tad convoluted, I'm open to suggestions for an alternate mechanism that accomplishes what is called out in https://github.com/envoyproxy/envoy/issues/11079 - barring that there are definitely some more docs required for this change.

Reply:
Right, my sense is that we should be able to do this without using a header value as control flow (and I would prefer to not use a header if possible).

For example, `statusOnError` and the other calls to sendLocalReply show precedent for setting the http status code to return to the downstream client. I bet we can just pass it below to decoder_callbacks_->sendLocalReply near L306.

Reply:
+1 for not using header if possible. But `statusOnError` is only honoured for http requests I guess because for gRPC it is overwritten https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.cc#L520 - So I think it is not possible without changing signature to let sendLocalReply in filter manager and utility know that it should not be overwritten,

Reply:
Yes after looking harder, I agree that sendLocalReply would need to be expanded to include the desired HTTP code for gRPC clients.

Reply:
So the consensus is that we should take the more explicit path of changing call signatures up the chain of `sendLocalReply()` variants to allow explicitly passing the HTTP response code?  
I will revert the changes to the V2 config and get started on replacing the header-magic with the more explicit path for V3+.

Reply:
I think sendLocalReply should support sending back the translated grpcToHttpStatus instead of just 200 OK. We can then configure the components that use sendLocalReply (ext_authz in this case, but in theory ratelimit as well, or even on the HttpConnectionManager which would apply the behavior to all downstreams for that HCM).

There are some signatures to modify going down, but eventually it becomes LocalReplyData which I think is a bit less painful to deal with.

Reply:
There are some mocks to modify if we change the signatures for the various `sendLocalReply()` functions, but it's relatively straightforward.  However, if we have `Utility::sendLocalReply()` call `grpcToHttpStatus()` there appear to be some dependency-cycle issues to get access to that function in grpc/common - any issue with just passing the HTTP code itself up the chain rather than a flag to tell `Utility::sendLocalReply()`  to figure out what code to use?

My first cut, which should be ready to push tomorrow, has the code passed along as an optional parameter until the `LocalReplyData` object is available, and then bundles it within that as a new variable - that seem sane to you?

Reply:
I would prefer to not add another http code parameter when one already exists.  I think a fine first cut would be to pass down a new boolean that says whether to use the http status for gRPC clients (no need for grpcToHttpStatus in this cut). This is similar to what you already have, but instead of the call to httpCodeForGrpcDeny it would just be checking that piped down parameter.

Reply:
Latest push adds that boolean, it's a new parameter for the steps before `LocalReplyData` is available.  Unit tests and manual via my modified ext_auth example folder are all passing.